src/node/mini_miner.h

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_NODE_MINI_MINER_H
+#define BITCOIN_NODE_MINI_MINER_H
+
+#include <txmempool.h>
+
+#include <memory>
+#include <optional>
+#include <stdint.h>
+
+namespace node {
+
+// Container for tracking updates to ancestor feerate as we include ancestors in the "block"
+class MockMempoolEntry
+{
+    CAmount fee_individual;
+    const CTransaction& tx;
```

Comment:
I'm not at all sure about this, but this is a reference variable so it's (sort of) a pointer, right? We continue to hold this reference after releasing `mempool.cs` -- is it possible for the transaction to go away (its memory deallocated) if, for example, it gets mined, thereby invalidating our reference? I wonder if this might be better:
```suggestion
    const CTransaction tx;
```
The debugger shows `sizeof(node::MockMempoolEntry)` is only 40 bytes with `tx` being a reference, and 152 if it's not a reference, which makes sense because `sizeof(CTransaction)` is 120.

Or, maybe it should be `const CTransactionRef tx;` because then it's a shared pointer, so if the tx gets removed from the mempool, our reference will remain valid until we're done with it. The advantage of this would be we wouldn't be making a full copy of the transaction.